barretenberg/cpp/src/barretenberg/vm2/simulation/lib/raw_data_dbs.cpp

```diff
@@ -754,6 +754,9 @@ IndexedLeaf<NullifierLeafValue> PureRawMerkleDB::get_leaf_preimage_nullifier_tre
 SequentialInsertionResult<PublicDataLeafValue> PureRawMerkleDB::insert_indexed_leaves_public_data_tree(
     const PublicDataLeafValue& leaf_value)
 {
+    // Check for cancellation before write.
+    check_cancellation();
```

Comment:
nit: same naming nit for these, it's not clear this are actually catastrophic exceptions